examples/05-custom-schema/05-latex-block/LaTex.tsx

```diff
@@ -0,0 +1,233 @@
+          }
+          setTimeout(() => {
+            if (textareaRef.current) {
+              textareaRef.current?.focus()
+              textareaRef.current?.setSelectionRange(textareaRef.current.value.length, textareaRef.current.value.length);
+            }
+          })
+        }
+      }, [open]);
+
+      const handleChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+        const val = e.target.value;
+        const pos = props.getPos?.();
+        const node = props.node;
+        const view = editor._tiptapEditor.view;
+
+        const tr = view.state.tr
+        .replaceWith(pos, pos+node.nodeSize, view.state.schema.nodes.latex.create(
+          {
+            ...node.attrs,
+          },
+          val ? view.state.schema.text(val) : null
+        ));
+
+        view.dispatch(tr);
+      }
+
+      return (
+        <NodeViewWrapper as={'span'}>
+          <Components.Generic.Popover.Root>
+            <Components.Generic.Popover.Trigger>
+              <span className={"latex"} ref={contentRef}>
+                {loading ? (
+                  <span className={"latex-loading"}>latex loading...</span>
+                ) : (
+                  <span className={"latex-content"} dangerouslySetInnerHTML={{ __html: html }}></span>
+                )}
+              </span>
+            </Components.Generic.Popover.Trigger>
+            <Components.Generic.Popover.Content
+              className={"bn-popover-content bn-form-popover"}
+              variant={"form-popover"}>
+              <textarea ref={textareaRef} className={"latex-textarea"} value={content} onChange={handleChange} />
+            </Components.Generic.Popover.Content>
+          </Components.Generic.Popover.Root>
+        </NodeViewWrapper>
+      );
+    };
+
+    return (props) => {
+      if (!(props.editor as any).contentComponent) {
+        return {};
+      }
+      const ret = ReactNodeViewRenderer(BlockContent, {
+        stopEvent: () => true,
+      })(props) as NodeView;
+      ret.setSelection = (anchor, head) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (ret as any).renderer.updateProps({
+          selectionHack: { anchor, head },
+        });
+      };
+
+      (ret as any).contentDOMElement = undefined;
+
+      const oldUpdated = ret.update!.bind(ret);
+      ret.update = (node, outerDeco, innerDeco) => {
+        const retAsAny = ret as any;
+        let decorations = retAsAny.decorations;
+        if (
+          retAsAny.decorations.decorations !== outerDeco ||
+          retAsAny.decorations.innerDecorations !== innerDeco
+        ) {
+          decorations = {
+            decorations: outerDeco,
+            innerDecorations: innerDeco,
+          };
+        }
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return oldUpdated(node, decorations, undefined as any);
+      };
+      return ret;
+    };
+  };
+  return nodeView;
+}
+
+const propSchema = {
```

Comment:
I don't think `open` should be part of the blockschema. It's a state of the UI, not of the document.

For example, this will mean that the open state will be stored in databases or synced over collaboration, I don't think that's what we want